src/commands/slash/voice/play/url.ts

```diff
@@ -0,0 +1,106 @@
+import { cmdEvent } from "@shared/interactions/interfaces";
+import { MusicSubscription } from "@shared/voice/subscription";
+import { Track } from "@shared/voice/track";
+import { getSubscription } from "@shared/voice/utils";
+import { ApplicationCommandData } from "discord.js";
+import { YT } from '@voice/youtubeapi'
+
+export const data: ApplicationCommandData = {
+    name: 'url',
+    description: 'Plays a song or a playlist from url. Supports YT only right now.',
+    options: [{
+        name: 'input',
+        description: 'the input url link',
+        type: 'STRING',
+        required: true
+    }]
+}
+
+export async function run(e: cmdEvent) {
+
+    let subscription: MusicSubscription
+    try {
+        subscription = await getSubscription(e);
+    }
+    catch (error) {
+        if (error instanceof Error) {
+            e.interaction.reply(error.message);
+        }
+        console.log(error);
+        return;
+    }
+
+    // Define responses for this interaction.
+    const eventMethods = {
+        onStart(title: string) {
+            e.interaction.followUp({ content: `Now playing **${title}**!` }).catch(console.warn);
+        },
+        onFinish(title: string) {
+            e.interaction.followUp({ content: `Now finished **${title}**!`, ephemeral: true }).catch(console.warn);
+        },
+        onError(title: string, error: Error) {
+            console.warn(error);
+            e.interaction.followUp({ content: `**${title}** error: ${error.message}` }).catch(console.warn);
+        },
+    }
+
+
+
+    const urlInput = e.interaction.options.get('input')!.value! as string;
+
+    // Regex: https://regexr.com/62j9j
+    const isYtVid = urlInput.match(/((http(s)?:)(\/\/))?(www\.)?((m.)?youtube.com|youtu.be)\/(watch)?(\?v=)?(\S+)?/) != null;
+
+    // This is defined only for YT videos
```

Comment:
What is this comment refering to?

Reply:
This comment is a leftover, will remove.